akd/src/directory.rs

```diff
@@ -65,11 +65,12 @@ where
     pub async fn new(storage: StorageManager<S>, vrf: V) -> Result<Self, AkdError> {
         let azks = Directory::<TC, S, V>::get_azks_from_storage(&storage, false).await;
 
-        if azks.is_err() {
-            // generate a new azks if one is not found
-            let azks = Azks::new::<TC, _>(&storage).await?;
-            // store it
-            storage.set(DbRecord::Azks(azks.clone())).await?;
+        if let Err(AkdError::Storage(StorageError::NotFound(_))) = azks {
+            // generate + store a new azks only if one is not found
+            let new_azks = Azks::new::<TC, _>(&storage).await?;
+            storage.set(DbRecord::Azks(new_azks.clone())).await?;
+        } else {
+            let _res = azks?;
```

Comment:
Maybe we can just do a single match statement on `azks?`:
```
match azks? {
  Ok(_) => Ok(Directory { ... }),
  Err(StorageError) => ... ,
  Err(e) => Err(e) ,
}
```

Thoughts?

Reply:
Works for me! We mostly care about the `StorageError::NotFound` occurrence here, so felt an if-let was a bit more succinct. That said, I don't have any strong reservations around using a match.

Reply:
For posterity -- @kevinlewi  and I spoke offline and felt that the if-let approach was a bit cleaner after some more discussion.